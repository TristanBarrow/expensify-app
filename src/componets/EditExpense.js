import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/dashboard');
  }
  onClick = () => {
    this.props.removeExpense(this.props.expense.id);
    this.props.history.push('/dashboard');
  }
  render() {
    return (
      <div>
        <div className='page-header'>
          <div className='content-container'>
              <h1 className='page-header__title'>Edit Expense</h1>
          </div>
        </div>
        <div className='content-container'>
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
            submitButton={'Save Expense'}
          />
          <button className='button button--gray' onClick={this.onClick}>Remove Expense</button>
        </div>
      </div>
    )
  }
};

const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  removeExpense: (id) => dispatch(startRemoveExpense(id))
});

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);