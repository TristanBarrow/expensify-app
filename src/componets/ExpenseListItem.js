import React from 'react';
import { Link } from 'react-router-dom'; 

const ExpenseListItem = ({ description, amount, createdAt, id }) => (
  <div>
    <Link to={`/edit/${id}`}>    
      <h3>{description}</h3>
    </Link>
    <p>
      Amount: {amount} <br />
      Time: {createdAt}
    </p>
  </div>
);

export default ExpenseListItem;