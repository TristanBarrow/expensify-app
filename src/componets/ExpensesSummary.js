import React from 'react';
import { connect } from 'react-redux';
import getVisibleExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = (props) => (
  <div>
    Viewing {props.expenseCount} expense{props.expenseCount === 1 ? '' : 's'
  } for a total of {
      numeral(props.expensesTotal / 100).format('$0,0.00')}.
  </div>
);

const mapStateToProps = (state) => ({
    expenseCount: getVisibleExpenses(state.expenses, state.filters).length,
    expensesTotal: getExpensesTotal(
      getVisibleExpenses(
        state.expenses, 
        state.filters
      )
    )
  });

export default connect(mapStateToProps)(ExpenseSummary);