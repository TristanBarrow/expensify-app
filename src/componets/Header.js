import React from 'react';
import { NavLink } from 'react-router-dom'; 

const Header = () => (
  <div>
    <h1>Expensify</h1>
    <NavLink activeClassName="is-active" to="/" exact={true}>Home Page</NavLink>
    <NavLink activeClassName="is-active" to="/create">Add Expense</NavLink>
  </div>
);
// <NavLink activeClassName="is-active" to="/help">Help</NavLink>

export default Header;