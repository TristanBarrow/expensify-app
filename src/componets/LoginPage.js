import React from 'react';

export class LoginPage extends React.Component {
  onClick() {
    console.log('You are Logging in...');
  };
  render() {
    return (
      <div>
        <h1>Wecome to Expensify</h1>
        <button onClick={this.onClick}>Login</button>
      </div>
    )
  }
};