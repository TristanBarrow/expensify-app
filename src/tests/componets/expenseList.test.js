import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../componets/ExpenseList';
import expenses from '../fixtures/expenses';

test('Render expense list with test values', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />); 
  expect(wrapper).toMatchSnapshot();
});

test('Render empty expense list', () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />); 
  expect(wrapper).toMatchSnapshot();
});