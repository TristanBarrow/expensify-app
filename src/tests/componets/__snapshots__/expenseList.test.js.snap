// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render empty expense list 1`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;

exports[`Render expense list with test values 1`] = `
<div>
  <ExpenseListItem
    amount={100}
    createdAt={-2678400000}
    description="AAA"
    id="ID 1"
    key="ID 1"
    note="Test Note"
  />
  <ExpenseListItem
    amount={300}
    createdAt={0}
    description="ABC"
    id="ID 2"
    key="ID 2"
    note="Test Note"
  />
  <ExpenseListItem
    amount={200}
    createdAt={2678400000}
    description="CCC"
    id="ID 3"
    key="ID 3"
    note="Test Note"
  />
</div>
`;
